service/src/test/java/uk/gov/hmcts/reform/fpl/utils/IncrementalIntegerTest.java

```diff
@@ -0,0 +1,19 @@
+package uk.gov.hmcts.reform.fpl.utils;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class IncrementalIntegerTest {
+    @Test
+    void shouldGetAndIncrementCurrentIntegerValue() {
+        IncrementalInteger i = new IncrementalInteger();
+        assertThat(i.getAndIncrement()).isZero();
```

Comment:
The test doesn't check if the increment happened.

I can implement return value; from th implementation and this test will always pass.